### PR TITLE
fix `det_pylibdir` provided by `PythonPackage` easyblock for Python 3.12+

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

In Python 3.12 the deprecated `distutils` module was removed but `det_pylibdir` requires `distutils.sysconfig`. There is a new `sysconfig` module which has most of its functionality as documented in https://peps.python.org/pep-0632/#migration-advice

However `sysconfig.get_python_lib` doesn't exist. But [`sysconfig.get_path`](https://docs.python.org/3/library/sysconfig.html#sysconfig.get_path) does. But it isn't clear how to replace the former with the latter.

As we mostly need this function for `pip install --prefix <foo>` I checked the pip sources and found relevant test code: https://github.com/pypa/pip/blob/50e3c500bd2b27dd72317cfd5ff5bb498540e416/tests/functional/test_install.py#L1322

So basically we switch
```
$ python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(plat_specific=False, prefix="/tmp/"))'
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
/tmp/lib/python3.10/site-packages

$ python -c 'import sysconfig; print(sysconfig.get_path("platlib", vars={"platbase": "/tmp/", "base": "/tmp/"}))'
/tmp/lib/python3.12/site-packages
```

### CODE REVIEW

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _tqdm-4.66.1-GCCcore-12.3.0.eb_ 
 * **SUCCESS** _matplotlib-3.7.2-gfbf-2023a.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3114.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/de699da49164f8e3bd3fb0d73c0a9433 for a full test report.
